### PR TITLE
Enhance some debug utils

### LINKED ISSUE #4646: Breakdown the "dbms/src/Storages/Transaction" module
## Enhancement
The codes under "dbms/src/Storages/Transaction" are some codes we adapt TiFlash with TiDB. After years of development, it become kind of crowded and confusing.

Suggest breakdown "dbms/src/Storages/Transaction” into these modules:
* Some global context for TiFlash management
* Basic structure from TiDB (TableInfo/DatabaseInfo, etc)
* Syncing schema with TiDB
* Decoding data from row to column (Block)
* Raft layer
  * Region info
  * Handling admin command
  * Handling write command
  * Handling apply snapshot/ingest sst
  * Handling learner read

<hr/>

1. Suggest creating a `dbms/src/TiDB` as the module for basic components from TiDB. And
    * [ ] Rename TMTContext to TiFlashContext
    * [ ] Move TableInfo/DBInfo, etc from TiDB.h to `dbms/src/TiDB`
    * [ ] Move TiFlashContext/ManagedStorages to the `dbms/src/TiDB` module
    * [ ] Add a `dbms/src/TiDB/Codec` and move the logic of decoding data from row to column there
    * [x] Add a `dbms/src/TiDB/Schema` and move schema syncing logic there
2. Rename `dbms/src/Transaction` to `dbms/src/KVStore`. And maybe we should further break down the codes for different Raft processes under this module.
    * [x] Rename `dbms/src/Transaction` to `dbms/src/KVStore` https://github.com/pingcap/tiflash/pull/8073
    * [x] Break down `KVStore`. https://github.com/pingcap/tiflash/pull/8073

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #4646

Problem Summary:

1. Make mock proxy's persistence pattern more similar to reality
2. Remove and aggregate abundant initializing procedures
3. Clear some codes in MockRaftStoreProxy
4. Introduce MockFFIImpls(originally in MockRaftStoreProxy.cpp) and MockReadIndex(originally in class MockRaftStoreProxy)
5. Add extra msg for RegionException

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/KVStore/tests/kvstore_helper.h [hunk 1]
```diff
@@ -114,8 +113,11 @@ class RegionKVStoreTest : public ::testing::Test
         kvstore = std::make_shared<KVStore>(global_ctx);
         // only recreate kvstore and restore data from disk, don't recreate proxy instance
         kvstore->restore(*path_pool, proxy_helper.get());
+        proxy_instance->reload();
+        global_ctx.getTMTContext().getRegionTable().clear();
         return *kvstore;
     }
+    // Onlt handle mock proxy's part. Conflicts with `debugAddRegions`.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/tests/kvstore_helper.h:120]:
```suggestion
    // Only handle mock proxy's part. Conflicts with `debugAddRegions`.
```
`Only`?

File: dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp [hunk 2]
```diff
@@ -195,12 +195,20 @@ void LearnerReadWorker::recordReadIndexError(RegionsReadIndexResult & read_index
...
         {
             const auto & region_error = resp.region_error();
             auto region_status = RegionException::RegionReadStatus::OTHER;
             if (region_error.has_epoch_not_match())
+            {
+                extra_msg = fmt::format(
+                    "read_index_resp {} epoch {}:{}",
+                    region_id,
+                    kvstore->getRegion(region_id)->version(),
+                    kvstore->getRegion(region_id)->confVer());
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp:209]:
```suggestion
                extra_msg = fmt::format(
                    "read_index_resp error, region_id={} version={} conf_version={}",
                    region_id,
                    kvstore->getRegion(region_id)->version(),
                    kvstore->getRegion(region_id)->confVer());
```

File: dbms/src/Storages/KVStore/Read/LearnerRead.cpp [hunk 3]
```diff
@@ -79,6 +79,7 @@ void validateQueryInfo(
 {
     RegionException::UnavailableRegions fail_region_ids;
     RegionException::RegionReadStatus fail_status = RegionException::RegionReadStatus::OK;
+    std::string fail_extra_msg;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/LearnerRead.cpp:82]: I think "err_msg" or "error_msg" is more native speaking in English

File: dbms/src/Storages/KVStore/Read/LearnerRead.cpp [hunk 4]
```diff
@@ -103,17 +105,19 @@ void validateQueryInfo(
             fail_status = status;
             LOG_WARNING(
                 log,
-                "Check after snapshot acquired from storage, region_id={} version={} handle_range={} status={}",
+                "Check after snapshot acquired from storage, region_id={} version={} handle_range={} status={} "
+                "fail_msg={}",
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/LearnerRead.cpp:109]:
```suggestion
                "err_msg={}",
```

File: dbms/src/Storages/KVStore/Read/LearnerRead.cpp (hunk 3 above)
Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/Read/LearnerRead.cpp:82]: The previous variables are all begin with `fail`

File: dbms/src/Storages/KVStore/Read/LearnerRead.cpp (hunk 4 above)
Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/Read/LearnerRead.cpp:109]: This message is extra, the final error msg is construncted in UnavailableRegions

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**